rust/connlib/snownet/src/node.rs

```diff
@@ -261,18 +257,13 @@ where
 
             c.state.on_upsert(cid, &mut c.agent, now);
 
-            for candidate in c.agent.local_candidates() {
-                signal_candidate_to_remote(cid, candidate, &mut self.pending_events);
-            }
-
-            // Server-reflexive candidates are not in the local candidates of the ICE agent so those need special handling.
-            for candidate in self
-                .shared_candidates
-                .iter()
-                .filter(|c| c.kind() == CandidateKind::ServerReflexive)
-            {
-                signal_candidate_to_remote(cid, candidate, &mut self.pending_events);
-            }
+            seed_agent_with_local_candidates(
+                cid,
+                c.relay,
+                &mut c.agent,
+                &self.allocations,
+                &mut self.pending_events,
+            );
```

Comment:
I've decided to just re-seed the client here. This is effectively the same as we did before because adding the same candidate again to the agent is a no-op and this allows me to reuse this function.